### PR TITLE
Documentation clean up, update to Documenter 1.0, add *.md to formatter (PSY portion)

### LINKED ISSUE #1196: docs: links to FuelCurve and CostCurve broken
Links to FuelCurve and CostCurve are broken in the docs. See the link on variable parameter in ThermalGenerationCost here:
https://nrel-sienna.github.io/PowerSystems.jl/stable/model_library/thermal_generation_cost/

### PR DESCRIPTION
**Pull request by kdayday**
- Update to documenter 1.0 with settings to error during compilation for everything except document size and missing docstrings
- Add *.md files to formatter script and run formatter on documentation
- Update dynamics data tutorial by merging with another tutorial in PSID
- resolve missing docstrings list
    - adding missing PSY files to API (mostly dynamics devices)
    - removing IS from `makedocs`
    - selecting only the IS functions/types that are exported by PSY through a new filter function
- make cost doc strings visible again after migration to IS
- Make JSON serialization how-to have a realistic folder name instead of temp folder
- expanded explanation on Per unit, which needs a review from someone else
- resolve other asks from feedback from first documentation rewrite, including moving text and fixing/removing some examples that were erroring

Successful compilation with documenter 1.0 depends on docstring updates in IS https://github.com/NREL-Sienna/InfrastructureSystems.jl/pull/394

Does not include moving the model library into API 

Closes #1196

### CODE REVIEW

**Comment by kdayday** [comment]
@jd-lara All checks are passing now, ready for review/merging

**Review by rodrigomha — APPROVED** [review]
The changes are too large to review. In a general glance they look really good, so I would recommend merging and solving issues if they are found later.

File: docs/src/api/glossary.md [hunk 1]
```diff
@@ -1,156 +1,135 @@
...
+    For contrast, see the definition for [Dynamic](@ref D) data.
+  - *STAB*: Speed Sensitive Stabilizing PSS Model
 
 ### V
 
-* *VSCDCLine*: Voltage-Source Converter Direct Current Line
+  - *VSCDCLine*: Voltage-Source Converter Direct Current Line
 
-* *VSM*:
+  - *VSM*:
```
Comment (jd-lara) [inline docs/src/api/glossary.md:119]: missing here

**Review by jd-lara — CHANGES_REQUESTED** [review]

File: docs/src/api/glossary.md [hunk 2]
```diff
@@ -1,156 +1,135 @@
...
+    intervals are represented using `Dates.Period` types. For instance, in a Day-Ahead market
+    simulation, the interval is usually `Hour(24)`.
+  - *Initial time*: The first time-stamp in a forecast window. See the article on
+    [`Time Series Data`](@ref ts_data).
+  - *IPC*: Interconnecting power converter
 
 ### O
 
-* *OEL*:
+  - *OEL*:
```
[the remark this answers is not in the document]
Comment (rodrigomha) [reply docs/src/api/glossary.md:73]: OEL: Over Excitation Limiter

File: docs/src/api/glossary.md (hunk 1 above)
Comment (rodrigomha) [reply docs/src/api/glossary.md:119]: VSM: Virtual Synchronous Machine

File: docs/src/api/glossary.md [hunk 3]
```diff
@@ -1,156 +1,135 @@
...
-
-* *PSID*: 
-
-* *PSLF*:
-
-* *pu* or *p.u.*: Per-unit
+  - *PSS*: Power System Stabilizer
+  - *PSSE* or *PSS/E*: Siemen's PSS®E Power Simulator
+  - *PPA*: Power purchase agreement
+  - *PSID*:
```
Comment (jd-lara) [inline docs/src/api/glossary.md:82]: also missing

File: docs/src/api/glossary.md [hunk 4]
```diff
@@ -1,156 +1,135 @@
...
+  - *PSLF*:
```
Comment (jd-lara) [inline docs/src/api/glossary.md:83]: missing

File: docs/src/api/glossary.md [hunk 5]
```diff
@@ -1,156 +1,135 @@
...
+  - *pu* or *p.u.*: Per-unit
 
 ### R
 
-* *REECB1*:
-
-* *REPCA1*:
+  - *REECB1*:
```
Comment (jd-lara) [inline docs/src/api/glossary.md:88]: missing

File: docs/src/api/glossary.md [hunk 6]
```diff
@@ -1,156 +1,135 @@
...
 
-* *Resolution*: The period of time between each discrete value in a time series. All resolutions
-  are represented using `Dates.Period` types. For instance, a Day-ahead market data set usually
-  has a resolution of `Hour(1)`, a Real-Time market data set usually has a resolution of `Minute(5)`. 
+  - *REPCA1*:
```
Comment (jd-lara) [inline docs/src/api/glossary.md:90]: missing

File: docs/src/api/glossary.md (hunk 3 above)
Comment (rodrigomha) [reply docs/src/api/glossary.md:82]: PSID: PowerSimulationsDynamics.jl

File: docs/src/api/glossary.md (hunk 4 above)
Comment (rodrigomha) [inline docs/src/api/glossary.md:83]: PSLF: Positive Sequence Load Flow (GE Software)

File: docs/src/api/glossary.md [hunk 7]
```diff
@@ -1,156 +1,135 @@
...
-
-* *PPA*: Power purchase agreement
-
-* *PSID*: 
-
-* *PSLF*:
-
-* *pu* or *p.u.*: Per-unit
+  - *PSS*: Power System Stabilizer
+  - *PSSE* or *PSS/E*: Siemen's PSS®E Power Simulator
```
Comment (rodrigomha) [inline docs/src/api/glossary.md:80]: Siemens' PSS®E Power System Simulator for Engineering

File: docs/src/api/glossary.md (hunk 5 above)
Comment (rodrigomha) [inline docs/src/api/glossary.md:88]: REECB1: Renewable Energy Electric Controller Type B1

File: docs/src/api/glossary.md (hunk 6 above)
Comment (rodrigomha) [inline docs/src/api/glossary.md:90]: REPCA1: Renewable Energy Power Controller Type A1

File: docs/src/explanation/dynamic_data.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
+# Dynamic Devices
+
+## Static and Dynamic Data Layers
+
+`PowerSystems.jl` uses two data layers to define data for dynamic simulations:
```
Comment (jd-lara) [inline docs/src/explanation/dynamic_data.md:5]:
```suggestion
`PowerSystems.jl` uses two categories to define data for dynamic simulations:
```

File: docs/src/explanation/dynamic_data.md
Comment (jd-lara) [inline docs/src/explanation/dynamic_data.md:1]: @rodrigomha who wrote this?

Comment (rodrigomha) [reply docs/src/explanation/dynamic_data.md:1]: probably me lol

File: docs/src/explanation/dynamic_data.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+ 1. [Static](@ref S) components, which includes the data needed to run a power flow problem
+ 2. [Dynamic](@ref D) components are those that define differential equations to run a transient simulation. These dyanamic
```
Comment (rodrigomha) [inline docs/src/explanation/dynamic_data.md:8]:
```suggestion
 2. [Dynamic](@ref D) components are those that define differential equations to run a transient simulation. These dynamic
```

File: docs/src/tutorials/get_component_data.md [hunk 10]
```diff
@@ -55,3 +54,71 @@ bus1 = get_component(ACBus, sys, "nodeA")
 @show get_magnitude(bus1);
 nothing #hide
 ```
+
+# copied over from the System
```
Comment (rodrigomha) [inline docs/src/tutorials/get_component_data.md:58]: Maybe remove this?

File: src/parsers/pm_io/LICENSE.md
Comment (rodrigomha) [inline src/parsers/pm_io/LICENSE.md:1]: We should revert the change here. This is the LICENSE for PowerModels.

File: src/parsers/im_io/LICENCE.md
Comment (rodrigomha) [inline src/parsers/im_io/LICENCE.md:1]: Ditto here for the LICENSE

Comment (rodrigomha) [reply src/parsers/im_io/LICENCE.md:1]: Also, we should change the name from LICENCE to LICENSE

**Review by jd-lara — COMMENTED** [review]
We should mention that the tabular data parser is about to disappear.

File: docs/src/api/glossary.md (hunk 5 above)
Comment (rodrigomha) [reply docs/src/api/glossary.md:88]: REECB1: Renewable Energy Electric Controller Type B1

File: docs/src/api/glossary.md (hunk 6 above)
Comment (rodrigomha) [reply docs/src/api/glossary.md:90]: REPCA1: Renewable Energy Power Controller Type A1

File: docs/src/explanation/dynamic_data.md
Comment (jd-lara) [reply docs/src/explanation/dynamic_data.md:1]: There is some crazy text there lol

File: docs/src/tutorials/get_component_data.md (hunk 10 above)
Comment (kdayday) [reply docs/src/tutorials/get_component_data.md]: This tutorial is a to-do item and isn't currently visible to viewers. There are multiple comments in there that need to go, but I removed this one now.

**Comment by kdayday** [comment]
> We should mention that the tabular data parser is about to disappear.

I split out the parsing how-to guide into separate topics, including a separate page on the tabular data parser with a warning at the top.

**Comment by kdayday** [comment]
> ## [Codecov](https://app.codecov.io/gh/NREL-Sienna/PowerSystems.jl/pull/1176?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=NREL-Sienna) Report
> 
> Attention: Patch coverage is `33.33333%` with `2 lines` in your changes missing coverage. Please review.
> 
> > Project coverage is 84.61%. Comparing base [(`3ce7468`)](https://app.codecov.io/gh/NREL-Sienna/PowerSystems.jl/commit/3ce7468c850745dccb9de4ce3adfd45bed7fe02e?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=NREL-Sienna) to head [(`be03b71`)](https://app.codecov.io/gh/NREL-Sienna/PowerSystems.jl/commit/be03b71e8ec4b49b897481bc9a444381752ea5d5?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=NREL-Sienna).
> > Report is 19 commits behind head on main.
> 
> [Files with missing lines](https://app.codecov.io/gh/NREL-Sienna/PowerSystems.jl/pull/1176?dropdown=coverage&src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=NREL-Sienna) 	Patch % 	Lines
> [src/models/OuterControl.jl](https://app.codecov.io/gh/NREL-Sienna/PowerSystems.jl/pull/1176?src=pr&el=tree&filepath=src%2Fmodels%2FOuterControl.jl&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=NREL-Sienna#diff-c3JjL21vZGVscy9PdXRlckNvbnRyb2wuamw=) 	33.33% 	[2 Missing ⚠️ ](https://app.codecov.io/gh/NREL-Sienna/PowerSystems.jl/pull/1176?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=NREL-Sienna)
> Additional details and impacted files

I think the two setters I added are covered by `test_accessors` already

File: docs/src/explanation/dynamic_data.md
Comment (kdayday) [reply docs/src/explanation/dynamic_data.md:1]: Anything else to change?

**Review by rodrigomha — APPROVED** [review]
I'm happy with this PR to be merged

**Review by jd-lara — APPROVED** [review]
